axelor-budget/src/main/java/com/axelor/apps/budget/db/repo/GlobalBudgetManagementRepository.java

```diff
@@ -32,4 +33,16 @@ public GlobalBudget copy(GlobalBudget entity, boolean deep) {
 
     return copy;
   }
+
+  @Override
+  public GlobalBudget save(GlobalBudget entity) {
+
+    GlobalBudgetService globalBudgetService = Beans.get(GlobalBudgetService.class);
+
+    globalBudgetService.fillGlobalBudgetOnBudget(entity);
```

Comment:
same issue as the latest PR on budget, this is a circular dependency because globalBudgetService has too many methods. see comments below to fix this